### DISCUSSION TITLE
Challenge definition

### DISCUSSION THREAD

**Discussion by eor314** [NOC-AI-Research/AISIT_challenge#4] [Ideas]
There are few bits we need to flesh out to formulate the competition text.

## Science goals
To define the challenge, we need to start by articulating the high level science goals. It sounds like the goal is to *track changes in meltwater patterns over time and identify drivers of variability*. Getting there requires constructing fields from sparse measurements that allow us to look for those patterns in freshwater circulation in the Arctic. Is that the right way to describe the ultimate desired science output?

## Impact on challenge definite
Once the science goal is settled on, we need to determine:
- What is the desired intermediate product (i.e. what we're asking participants to make) that will allow us to answer the science questions?
- What will we use for evaluation data?
- Can we bring in other datasets to frame this as a multimodal problem?

**Comment by eor314** [comment]
@Kzra @lupinthief @arcable96 @thopri @jdconey 

I started this discussion to collect ideas and thoughts on the challenge definition following on from the meeting earlier today. I think I've written out the big picture issues we identified. My sense is we use Tom and Archie's models as baselines, build from Ezra's proposed text (in this [Markdown](https://github.com/NOC-AI-Research/AISIT_challenge/blob/main/ezra-gridded-challenge.md)), and slot material into the pages Jon created.

**Comment by arcable96** [comment]
This is a great idea. From my point of view, I think the idea of generating a gridded dataset as an outcome is a good foundation. The main issue that I see with this is having a clear achievable target for the challengers. 

For clarity, the process that Tom and I have been using is as follows:
1. Take 90% of the BAS dataset and train it on the d18O data using linearised ML model, with salinity, temperature, depth, longitude & latitude as variable inputs
2. The final 10% of the data is then used for inference 
3. Apply the weights from the ML model to the Arctic Ocean Reanalysis (AOR) dataset (which has gridded, spatially-complete salinity & temperature data) to get a gridded, spatially-complete d18O dataset (call this the AOR ML d18O dataset).

There are variations of this we've done (e.g. remove a year's worth of BAS data at the outset to use for testing at the end) but this is the overall method. However, evaluating it in a succinct way is tricky. Here are a couple of ideas I have on this.

- Use statistical evaluation tools e.g. RMSE. Compute the RMSE of the BAS and AOR salinity datasets where there is overlap and compare that with the equivalent RMSE of the BAS and AOR ML d18O datasets. If the RMSEs are similar, implies the ML is doing a good job (maybe?). Is there a way we can relate these two things in a succinct metric?
- Compute a physical metric e.g. relating to the Arctic Ocean Oscillation, that can be compare across different outputs. Main problem with this is I'm not sure exactly what this would be and there's not much time left on the project!

**Comment by thopri** [comment]
I may be doing that classic XKCD comic Jonanthon posted and over estimating what people outside of ocean science know but I think an AOO like index reconstruction would be awesome. It could be pitched as calcuting the "Arctic AMOC" as the AMOC has been in the news a lot so potentially fairly well known?

I would expect that product challengers make would be the gridded product and we apply a AOO reconstruction function and say can you do a better job than our baseline model? 

I do think we would also want to assess skill in a more traditional way e.g. RMSE but I don't have a good answer aside from holding some of the BAS dataset back which can be easily gamed as we can't partially release the database. Comparing RMSE's of important variables like salinity as Archie mentions above could be a good solution although it makes me worry people will chase overfitting that relationship to look better when they actually aren't better at o18 reconstruction.

**Comment by Kzra** [comment]
Thanks @eor314 for starting this discussion. 

1. I think the output should be a gridded dataset of O18 in the Arctic Ocean
2. For evaluation/training data, I think we give participants our O18 database as a csv file. We can optionally provide code (see reference notebook) to convert this into a gridded dataset without a time/depth dimension (and so participants can write their own code to add these extra dimensions to the dataset if they want). 
3. For additional datasets, I basically think we tell participants they can use whatever publicly accessible climate/water data they can find. The essential thing is that anyone can run their code. We optionally provide code to show how you can use the Arctic Ocean Reanalysis (AOR) dataset to get salinity data for the entire Arctic Ocean (thanks @arcable96 @thopri). But if I was doing the challenge, i'd also want to look at other reanalysis datasets that are out there. 

I'm a fan of us posing the challenge loosely, along the lines of:


> Develop a Python notebook which produces a gridded dataset of O18 isotope abundance across the Arctic ocean. Your notebook will be assessed on the following criteria: 
> 
> 1. Application of machine learning - can you demonstrate innovative machine learning techniques that differ from the approaches provided in the [reference notebook](https://github.com/[NOC-MDP/AISIT](https://github.com/NOC-MDP/AISIT/blob/main/case_study.ipynb)/blob/main/case_study.ipynb). 
> 2. Extensivity of the grid - can you create a 4D grid that copes with depth and encodes seasonality/time, in our [reference notebook](https://github.com/[NOC-MDP/AISIT](https://github.com/NOC-MDP/AISIT/blob/main/case_study.ipynb)/blob/main/case_study.ipynb) our grid is only 2D. 
> 3. Evaluation metrics - can you convince us that you understand how well your model fits the supplied training data. You'll be judged based on choice of metric (RMSE, R-Squared etc,) and how critically you assess the scores that you get using these metrics.
> 4. Application of the grid and computation of an AO18Index - can you generate ideas for how the grid could be applied in ocean science, for example, what correlations with other oceanographic processes you might want to look at? Additionally, do you have any thoughts on how your gridded data could be represented as a single daily/monthly index in a manner akin to how atmospheric gridded data is described by the Arctic/North Atlantic Oscillation Index?


We then supply a modified version of @arcable96 @thopri [notebook ](https://github.com/[NOC-MDP/AISIT](https://github.com/NOC-MDP/AISIT/blob/main/case_study.ipynb)/blob/main/case_study.ipynb) which is our template for how we want the entries to look. We judge based on how well the case is presented in the notebook as opposed to quantitatively on an evaluation score. 

For example, a team that concludes their grid isn't very accurate, but thoroughly presents a case as to why that is, would do better than a team that presents a high RMSE score but doesn't critically evaluate why that score might be misleading. Making the entrants choose their own evaluation metric is a bit of a free pass for us - but I think it's the right way to go, as we might get some good ideas that we otherwise might have missed.

**Reply by jdconey** [reply]
hmm 
It feels like we have a bit of decision to make!
If we go down the Kaggle route, then we'd most likely need an objective metric (we can create our own - maybe even a hybrid of RMSE and AOO) and a held out dataset that we don't share with participants that we can measure the submissions with.
Or we can go down something more loose as Ezra is suggesting - which I think has legs, but Kaggle might not be the right place to host it.

**Reply by Kzra** [reply]
Agreed! I think the notebook competition landing page would probably be hosted on the AISIT website, and we'd raise awareness of it through a series of in-person events (dare I say hackathons) and social media posts, etc. 

I think for the AISIT deliverables, we don't actually need to run the notebook competition. What I think might be feasible by the end of the project is to develop the competition materials (e.g. competition summary, website landing page design, reference notebooks, ideas for how to promote competition document) that could be included in the final AISIT report.

**Reply by jdconey** [reply]
Yes - I think we should outline two challenges in the report: one kaggle style that's purely data-driven and include the metric we want to evaluate against; and the second the more loose style.
The science text is looking good, and we can incorporate the notebooks that Archie and Tom have done already.
